### ISSUE TITLE
False positive: recur-argument count mismatch with core.async

### ISSUE THREAD

**Issue by gordonsyme** [borkdude/clj-kondo#422]
**version**

`clj-kondo v2019.08.01-alpha-SNAPSHOT` - supplied in https://github.com/borkdude/clj-kondo/issues/413

**platform**

macOS High Sierra

**problem**

Use of `recur` within `core.async/alt!!` (inside a `loop`) triggers a recur argument count mismatch error, it looks as though the `alt!!` bindings are being matched against rather than the `loop` bindings.

**repro**

This is as minimal a reproduction as I have gotten while still being working code.

```clojure
(ns clj-kondo-recur.core
  (:require [clojure.core.async :as async]))

(defn foo
  [chan]
  (loop []
    (async/alt!!
      (async/timeout 500)
      (do (println "timedout")
          (recur))

      chan
      ([v _ch]
       (println "got" v)
       (recur)))))

(defn go
  []
  (let [chan (async/chan)
        thread (future (foo chan))]
    (Thread/sleep 1000)
    (async/>!! chan "Hello!")
    (Thread/sleep 1000)
    (future-cancel thread)))
```

**expected behaviour**

Linting passes.

**actual behaviour**

clj-kondo emits this error:
```
/private/tmp/clj-kondo-recur/src/clj_kondo_recur/core.clj:16:8: error: recur argument count mismatch (expected 2, got 0)
```

**Comment by borkdude** [comment]
@gordonsyme Could this be a duplicate of https://github.com/borkdude/clj-kondo/issues/393?
The fix will be released soon.

**Comment by borkdude** [comment]
@gordonsyme Thanks. This repro works, I'll have a look soon.

**Comment by borkdude** [comment]
@gordonsyme This Mac binary should contain the fix: https://5405-176829714-gh.circle-artifacts.com/0/release/clj-kondo-2019.08.01-alpha-SNAPSHOT-macos-amd64.zip

**Comment by borkdude** [comment]
Fixed with https://github.com/borkdude/clj-kondo/commit/3fd613ff329669c70a9ecdea86e4686f2b01b5e2

**Comment by gordonsyme** [comment]
Thanks! I've run that binary against our source, confirmed the recur argument mismatch is no longer reported.
